debian/mariadb-server-10.5.postrm

```diff
@@ -13,20 +13,28 @@ MYADMIN="/usr/bin/mysqladmin --defaults-file=/etc/mysql/debian.cnf"
 
 # Try to stop the server in a sane way. If it does not success let the admin
 # do it himself. No database directories should be removed while the server
-# is running!
+# is running! Another mysqld in e.g. a different chroot is fine for us.
 stop_server() {
-  set +e
-  systemctl stop mysql
-  errno=$?
-  set -e
+    # Return immediately if there are no mysql processes running
+    # as there is no point in trying to shutdown in that case.
+    if ! pgrep -x mysqld > /dev/null; then return; fi
```

Comment:
Perhaps, but in this case one cannot just add the path or pgrep will stop returning any results.
```
otto@tre-test3:~$ pgrep -x mysqld
2168
otto@tre-test3:~$ pgrep -x /usr/sbin/mysqld
otto@tre-test3:~$ 
```

This is what has been in Debian downstream for a couple of years and has worked fine.

Reply:
What happens if the process is mariadbd? Shouldn't we test for that as well? I didn't check that this is actually possible, but I'll let @robertbindar test that.

Reply:
> What happens if the process is mariadbd? Shouldn't we test for that as well? I didn't check that this is actually possible, but I'll let @robertbindar test that.

This will not work if process is mariadbd. The packaging or init or systemd currently has no support for mariadbd as the process name (for now) is mysqld. This should be addressed when the process name changes and not "over-engineered" blindly in advance without being able to actually fully test it.